### DISCUSSION TITLE
grid-cols-[n] not working in NextJS.

### DISCUSSION THREAD

**Discussion by souls-edge** [tailwindlabs/tailwindcss#7041] [Help]
Good eve,

I have played around in a tailwind sandbox. Im upgraded, but lost confidence using it the moment i went into NextJS.

If i code this:
```

      <div className="grid grid-cols-[1fr_200px]">
        <div className="bg-[#777777]"> 1 </div>
        <div className="bg-sky-800"> 2 </div>
      </div>
```
Then the custom background color is shown but the custom grid-cols is not being invoked, the code is printed out to web   `<div className="grid grid-cols-">` has  been purged.

I am aware that  purgecss may be responsible, and also mentioned is a fix involving installing cross-env to resolve the issue.

looking at my package.json >    
 "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.1.4",
 "postcss": "^8.1.9",

**Comment by sarnielmorgia** [comment]
Looks ok to me. I'm using tailwindcss 3.0.12
![image](https://user-images.githubusercontent.com/91003451/148890652-2dfaa983-10fd-4770-a5e2-c9eb5667690b.png)

**Comment by simonswiss** [comment]
Hey!

You should not use the `postcss7-compat` package with Next.js - Next.js supports PostCSS 8, so you can use the normal version Tailwind CSS.

I'd replace your version of Tailwind with `npm install -D tailwindcss@latest 👍

What you're describing defniitely works with Next.js

**Comment by codepeach** [comment]
I've got the same issue. 
My package json file:

`  "devDependencies": {
    "autoprefixer": "^10.2.6",
    "postcss": "^8.3.5",
    "tailwindcss": "^3.0.15"
  }`

running npm install -D tailwindcss@latest didn't change it for me

**Comment by mostafa-rio** [comment]
@codepeach Did you manage to fix this? I think I'm getting kind of the smae issue

**Comment by pfmartins-zartis** [comment]
I was having a similar issue when trying to use classes based on props (dynamically assigned). 
It's definitely a bug.
A workaround I found is I created a scss file and applied the rules there and then used the scss file classes like follows:
```
/// grid.scss
.grid {
  &.container {
    display: grid;
    @for $i from 1 through 12 {
      @media (screen(sm)) {
        &.sm-gap-#{$i} {
          @apply gap-#{$i};
        }
        &.sm-#{$i}{
          @apply grid-cols-#{$i};
        }
        .item {
          &.sm-#{$i}{
            @apply col-span-#{$i};
          }
        }
      }

      @media (screen(md)) {
        &.sm-gap-#{$i} {
          @apply gap-#{$i};
        }
        &.sm-#{$i}{
          @apply grid-cols-#{$i};
        }
        .item {
          &.sm-#{$i}{
            @apply col-span-#{$i};
          }
        }
      }

      @media (screen(lg)) {
        &.sm-gap-#{$i} {
          @apply gap-#{$i};
        }
        &.sm-#{$i}{
          @apply grid-cols-#{$i};
        }
        .item {
          &.sm-#{$i}{
            @apply col-span-#{$i};
          }
        }
      }
    }
  }
}
```

```
//Grid.tsx
export function GridContainer({
  columns,
  children,
  spacing,
  ...props
}: GridContainerProps) {
  const columnsClasses = getColumnsClasses(columns);
  const spacingClasses = getSpacing(spacing);
  return (
    <div
      className={clsx(
        styles.grid,
        "w-full",
        styles.container,
        columnsClasses,
        spacingClasses
      )}
      {...props}
    >
      {children}
    </div>
  );
}

export function GridItem({ colspan, children }: GridItemProps) {
  const columnsClasses = getColumnsClasses(colspan, false);
  return (
    <div
      className={clsx(
        styles.grid,
        styles.item,
        columnsClasses,
      )}
    >
      <div>{children}</div>
    </div>
  );
}
```
I hope it helps. And I think this help issue should be turned into a bug.
